cmd/talosctl/main.go

```diff
@@ -6,17 +6,251 @@
+	reportCtxCancel()
+	wg.Wait()
+
+	rep.Report(reporter.Update{
+		Message: fmt.Sprintf("OpenAPI prompt succeeded for input %s.\nSuggestion of model %q (WARNING: USE IT AT YOUR OWN RISK):", inputStr, model),
+		Status:  reporter.StatusSucceeded,
+	})
+
+	fmt.Fprint(os.Stderr, "\n")   //nolint:errcheck
+	fmt.Fprint(os.Stderr, result) //nolint:errcheck
+	fmt.Fprint(os.Stderr, "\n")   //nolint:errcheck
+
+	if insufficientTokens {
+		fmt.Fprintf(os.Stderr, "<result is cut in half due to insufficient OPENAI API tokens (%d). Try increasing %s>\n", maxTokens, openAIMaxTokensEnvVar) //nolint:errcheck
+	}
+
+	return nil
+}
+
+func completion(ctx context.Context, client *openai.Client, model string, maxTokens int, inputJSON []byte) (string, bool, error) {
+	prompt := fmt.Sprintf("I ran talosctl and got an error. "+
+		"Help me to troubleshoot & fix it. Print steps. Be brief. Here's the info: %s\n", string(inputJSON))
+
+	req := openai.CompletionRequest{
+		Model:     model,
+		Prompt:    prompt,
+		MaxTokens: maxTokens,
+	}
+
+	resp, err := client.CreateCompletion(ctx, req)
+	if err != nil {
+		return "", false, fmt.Errorf("failed to create OpenAI completion: %w", err)
+	}
+
+	insufficientTokens := false
+
+	choices := slices.Map(resp.Choices, func(t openai.CompletionChoice) string {
+		if t.FinishReason == "length" {
+			insufficientTokens = true
+		}
+
+		return t.Text
+	})
+
+	if insufficientTokens {
+		choices = append(choices, fmt.Sprintf("<result is cut in half due to insufficient OPENAI API tokens. Try increasing %s (current: %d)>", openAIMaxTokensEnvVar, maxTokens))
+	}
+
+	return strings.TrimSpace(strings.Join(choices, "\n")), insufficientTokens, nil
+}
+
+func chatCompletion(ctx context.Context, client *openai.Client, model string, maxTokens int, inputJSON []byte) (string, bool, error) {
+	req := openai.ChatCompletionRequest{
+		Model: model,
+		Messages: []openai.ChatCompletionMessage{
+			{
+				Role:    "system",
+				Content: "You are a support agent helping Talos Linux users to help debug & fix their talosctl related issues. You reply with brief info in steps.",
+			},
+			{
+				Role:    "user",
+				Content: fmt.Sprintf("Here's the info I gathered as JSON: %s\n", string(inputJSON)),
+			},
+		},
+		MaxTokens: maxTokens,
+	}
+
+	resp, err := client.CreateChatCompletion(ctx, req)
+	if err != nil {
+		return "", false, fmt.Errorf("failed to create OpenAI completion: %w", err)
+	}
+
+	insufficientTokens := false
+
+	choices := slices.Map(resp.Choices, func(t openai.ChatCompletionChoice) string {
+		if t.FinishReason == "length" {
+			insufficientTokens = true
+		}
+
+		return t.Message.Content
+	})
+
+	return strings.TrimSpace(strings.Join(choices, "\n")), insufficientTokens, nil
+}
+
+func readTalosconfig() (map[string]any, error) {
+	talosConfig, err := config.Open("")
+	if err != nil {
+		return nil, err
+	}
+
+	for _, configContext := range talosConfig.Contexts {
+		configContext.CA = ""
+		configContext.Crt = ""
+		configContext.Key = ""
```

Comment:
might as well the tokens when using an omni downloaded talosctl

Reply:
Yes, but this is nothing to be reviewed to be merged, rather about the rough idea.